internal/pipeline/pipeline.go

```diff
@@ -119,6 +121,24 @@ func (p *Pipeline) Validate() error {
 	return nil
 }
 
+// fnChain returns a slice of function runners from the
+// functions and configs defined in pipeline.
+func (p *Pipeline) fnChain() ([]kio.Filter, error) {
+	fns := []Function{}
+	fns = append(fns, p.Generators...)
+	fns = append(fns, p.Transformers...)
+	fns = append(fns, p.Validators...)
```

Comment:
Do you currently treat validators differently from executors? Validators are not allowed to modify ResourceList.items

Leave TODO(s) for any functionality that's punted to future PR(s). Otherwise, it makes it hard to distinguish if something is a potential bug.

Reply:
TODO added.